src/exporter/DaemonMetricCollector.cc

```diff
@@ -157,6 +156,51 @@ void DaemonMetricCollector::dump_asok_metrics() {
         dump_asok_metric(perf_info, perf_values, name, labels);
       }
     }
+    // fetch labeled perf counters if config is set to true
+    bool labeledperf = g_conf().get_val<bool>("exporter_get_labeled_counters");
+    if (labeledperf) {
+       std::string counter_dump_response =
+        asok_request(sock_client, "counter dump", daemon_name);
+      if (counter_dump_response.size() == 0) {
+          failures++;
+          continue;
+      }
+      std::string counter_schema_response =
+          asok_request(sock_client, "counter schema", daemon_name);
+      if (counter_schema_response.size() == 0) {
+        failures++;
+        continue;
+      }
+
+      json_object counter_dump = boost::json::parse(counter_dump_response).as_object();
+      json_object counter_schema = boost::json::parse(counter_schema_response).as_object();
+
+      for (auto &labeled_perf : counter_schema) {
+        std::string labeled_perf_group = {labeled_perf.key().begin(), labeled_perf.key().end()};
+        json_object labeled_perf_group_object = labeled_perf.value().as_object();
+        auto counters = labeled_perf_group_object["counters"].as_object();
+        auto counters_labels = labeled_perf_group_object["labels"].as_object();
+        labels_t labels;
+
+        for(auto &label: counters_labels) {
+          std::string label_key = {label.key().begin(), label.key().end()};
+          labels["ceph_daemon"] = quote(daemon_name);
+          labels[label_key] = quote(label.value().as_string().c_str());
+        }
+        for (auto &counter : counters) {
+          json_object counter_group = counter.value().as_object();
+          if (counter_group["priority"].as_int64() < prio_limit) {
+            continue;        
+          }
+          std::string counter_name_init =  {counter.key().begin(), counter.key().end()};
+          std::string counter_name = "ceph_" + labeled_perf_group + "_" + counter_name_init;
+          std::replace_if(counter_name.begin(), counter_name.end(), is_hyphen, '_');
+
+          auto perf_values = counter_dump[labeled_perf_group].as_object()["counters"].as_object().at(counter_name_init);
```

Comment:
Same thing here, `counter_dump[labeled_perf_group].as_object()["counters"]` does not depend on anything inside the loop so I'd fetch it outside and then just:
```suggestion
          auto perf_values = counter_dump_labeled_blahblah.at(counter_name_init);
```
Keep in mind that this is something inside a for-for-for loop, so let's move to the outer scope whatever that is not strictly required in the inner loops.